crates/store/src/state.rs

```diff
@@ -610,7 +690,7 @@ impl State {
             .find_map(|(index, header)| {
                 (header.block_num() == batch_reference_block).then_some(index)
             })
-            .expect("DB should have returned the header of the batch reference block");
+            .ok_or(GetBatchInputsError::BatchRefernceBlockError)?;
```

Comment:
```suggestion
            .expect("DB should have returned the header of the batch reference block");
```

This should still be the expect and we can remove the `BatchRefernceBlockError` variant.